### PR TITLE
[dashboard] Authenticate with Public API using Cookies instead of Tokens

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Using existing Cookies to authenticate against Public API

## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/14398

## How to test

1. Open up preview
2. Sign in, open up console
3. Create Team, notice the request to HTTP `CreateTeam`, observe it's sending Cookies and receives an OK response

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/dashboard/src/service/public-api.ts
Comment (geropl) [inline components/dashboard/src/service/public-api.ts:16]: I assume all usage is featureFlagged up until now. Correct?

**Review by geropl — APPROVED** [review]
Tested and works. :heavy_check_mark: 

/hold bc of [this question](https://github.com/gitpod-io/gitpod/pull/14403/files#r1013746105), please unblock yourself if true :pray:

File: components/dashboard/src/service/public-api.ts
Comment (easyCZ) [reply components/dashboard/src/service/public-api.ts:16]: It still remains feature flagged. This just changes the "transport" layer such that the calls use cookies instead of generating a token.

Comment (easyCZ) [reply components/dashboard/src/service/public-api.ts:16]: Basically, this transport doesn't have any effect unless a call is actually made

**Pull request merged**
